jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtil.java

```diff
@@ -983,16 +1002,24 @@ protected static void mergeMetadata(HasMetadata item1, HasMetadata item2) {
      * when overriding
      */
     private static Map<String, String> mergeMapsAndRemoveEmptyStrings(Map<String, String> overrideMap, Map<String, String> originalMap) {
-        Map<String, String> answer = MapUtil.mergeMaps(overrideMap, originalMap);
-        Set<Map.Entry<String, String>> entries = overrideMap.entrySet();
-        for (Map.Entry<String, String> entry : entries) {
-            String value = entry.getValue();
-            if (value == null || value.isEmpty()) {
-                String key = entry.getKey();
-                answer.remove(key);
+        if (overrideMap == null && originalMap == null) {
+            return Collections.emptyMap();
+        } else if (originalMap == null) {
+            return overrideMap;
+        } else if (overrideMap == null) {
+            return originalMap;
```

Comment:
MapUtil.mergeMaps already takes care of the initial checks you've done here
https://github.com/eclipse/jkube/blob/645268c18c1e43bad18cb146362cf253900904e3/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/MapUtil.java#L59-L67

You probably just want to do this:
```java
final Map<String, String> answer = MapUtil.mergeMaps(overrideMap, originalMap);
if (overrideMap != null && originalMap != null) {
  /* ... */
}
return answer;
```